src/Headers.ts

```diff
@@ -0,0 +1,274 @@
+/// <reference path="./.d.ts"/>
+"use strict";
+
+export interface Header {
+  [name: string] : string;
+}
+
+export interface HeadersList extends Array<Header> {};
+
+// String based enum
+const Guards = {
```

Comment:
Better than strings all over the place. Could use enum but still need to look up strings, so meh.